gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/EnforceFlowFinishDeadlineDagProc.java

```diff
@@ -60,7 +60,9 @@ protected void enforceDeadline(DagManagementStateStore dagManagementStateStore,
         DagProcUtils.cancelDagNode(dagNodeToCancel, dagManagementStateStore);
       }
 
-      dag.setFlowEvent(TimingEvent.FlowTimings.FLOW_RUN_DEADLINE_EXCEEDED);
+      DagProcUtils.setAndEmitFlowEvent(eventSubmitter, dag, TimingEvent.FlowTimings.FLOW_CANCELLED);
```

Comment:
job level cancellation event is emitted from inside DagProcUtils.cancelDagNode
and then when Reevaluate dag proc is created because of this event, and that Reevaluate calls 
DagProcUtils.setAndEmitFlowEvent()

I think I've seen dag status being set to CANCELLED when it misses deadline.
Are you not seeing FLOW CANCELLED event?